lib/matplotlib/tests/test_datetime.py

```diff
@@ -20,11 +20,61 @@ def test_annotate(self):
         fig, ax = plt.subplots()
         ax.annotate(...)
 
-    @pytest.mark.xfail(reason="Test for arrow not written yet")
     @mpl.style.context("default")
     def test_arrow(self):
-        fig, ax = plt.subplots()
-        ax.arrow(...)
+        mpl.rcParams["date.converter"] = 'concise'
+        fig, (ax1, ax2, ax3) = plt.subplots(3, 1,
+                                            constrained_layout=True,
+                                            figsize=(10, 12))
+
+        start_date = datetime.datetime(2023, 1, 1)
+        dates = [start_date + datetime.timedelta(days=i) for i in range(31)]
+        y_values = range(31)
+
+        ax1.plot(dates, y_values, marker='o')
+        ax1.arrow(x=dates[10],
+                  y=10,
+                  dx=10,
```

Comment:
I would expect the `d[xy]` values that are used on date-based axes to be `timedelta` rather than raw numbers (and in particular if it _doesn't_ work that is something we should look at)

Reply:
Thanks for your review. Unfortunately, currently the function `ax.arrow` expects the `dx`, `dy` to be in the axis units only.  We have discovered another issue while implementing datetime test. 
Looks like this PR would be blocked meanwhile.

I have tested a minimal change here to get the timedelta working when         `dx=datetime.timedelta(days=10)` with the following patch. 
New unitless `d[xy]` are computed before passing down to the function `mpatches.FancyArrow`. The drawback is that we now need to import datetime object dependence into the `lib/matplotlib/axes/_axes.py`. Would be appreciated if you would suggest a better approach to implement the d[xy] timedelta support. 

```
diff --git a/lib/matplotlib/axes/_axes.py b/lib/matplotlib/axes/_axes.py
index 2136ecb2eb..780e5048e0 100644
--- a/lib/matplotlib/axes/_axes.py
+++ b/lib/matplotlib/axes/_axes.py
@@ -3,6 +3,7 @@ import itertools
 import logging
 import math
 from numbers import Integral, Number, Real
+import datetime
 
 import numpy as np
 from numpy import ma
@@ -5218,13 +5219,22 @@ default: :rc:`scatter.edgecolors`
         ...             arrowprops=dict(arrowstyle="->"))
 
         """
+        old_x=x
+        old_y=y
         # Strip away units for the underlying patch since units
         # do not make sense to most patch-like code
         x = self.convert_xunits(x)
         y = self.convert_yunits(y)
-        dx = self.convert_xunits(dx)
-        dy = self.convert_yunits(dy)
-
+        # if dx is timedelta, compute the unitless dx after convert
+        if isinstance(dx, datetime.timedelta):
+            dx = self.convert_xunits( old_x + dx ) - x
+        else:
+            dx = self.convert_xunits(dx)
+        # if dy is timedelta, compute the unitless dy after convert
+        if isinstance(dy, datetime.timedelta):
+            dy = self.convert_xunits( old_y + dy ) - y
+        else:
+            dy = self.convert_yunits(dy)
         a = mpatches.FancyArrow(x, y, dx, dy, **kwargs)
         self.add_patch(a)
         self._request_autoscale_view()
```